### PR TITLE
Java SDK release script

### PR DESCRIPTION
**Pull request by davidheryanto**
This pull request adds a script and ProwJob to release Feast Java SDK to **Maven Central** upon creation of new **release tags**.

I reverted the changes in https://github.com/gojek/feast/pull/394 where the revision value in `pom.xml` is provided explicitly. I use our previous approach where we set the `revision` property, which will then substitute `${revision}`placeholder in `pom.xml` in different modules. Reference: https://maven.apache.org/maven-ci-friendly.html

This is so that the **CI** script can more easily **override** the revision from external value e.g. from latest **Git tag**, like so `mvn -Drevision=$GIT_TAG`. This approach ensures that Git tag is the source of truth of the released version number. Updating the `<version>` tag manually on all `pom.xml` upon new version release is arguably more error prone e.g. we miss one file by accident.

I add **flatten-maven-plugin** that generates a flattened version of `pom.xml` during deployment. This is useful since our Java SDK depends on the parent module where the version is provided via `${revision}` variable. This flattened pom ensures that these dependencies can be resolved.

I also add **maven-gpg-plugin** to sign the components. This is required when releasing libraries to Maven Central.

### CODE REVIEW

**Comment by woop** [comment]
@davidheryanto thanks for this!

Can I ask that we move these scripts to `infra/scripts` or something similar?

**Comment by davidheryanto** [comment]
Okay sure.
I'll move the CI script into infra folder in another pull request so it's easier to track the changes.

File: pom.xml [hunk 1]
```diff
@@ -35,6 +35,7 @@
     </modules>
 
     <properties>
+        <revision>0.3.6-SNAPSHOT</revision>
```
Comment (zhilingc) [inline pom.xml]: should this be 0.4.2-SNAPSHOT?

Comment (davidheryanto) [reply pom.xml]: Updated

**Comment by ches** [comment]
Should this be backported for publishing v0.3 SDK?

**Comment by ches** [comment]
Actually maybe it'd be easiest for me to bring it into #407?

**Comment by davidheryanto** [comment]
@ches yes sounds good we can backport it in #407 
Thanks

**Pull request merged**